x-pack/plugins/global_search_bar/public/search_syntax/query_utils.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Query } from '@elastic/eui';
+import { FilterValues } from './types';
+
+export const getFieldValueMap = (query: Query) => {
```

Comment:
Could you add a comment explaining what this function does? It's not super obvious from the name or implementation.